core/rlwe/keys.go

```diff
@@ -417,6 +417,110 @@ func (evk EvaluationKey) Expand(params ParameterProvider, buffer *GadgetCipherte
 	return nil
 }
 
+// BinarySize returns the serialized size of the object in bytes.
+func (evk EvaluationKey) BinarySize() (size int) {
+	return evk.GadgetCiphertext.BinarySize() + len(evk.Seed)
+}
+
+// WriteTo writes the object on an [io.Writer]. It implements the [io.WriterTo]
+// interface, and will write exactly object.BinarySize() bytes on w.
+//
+// Unless w implements the [buffer.Writer] interface (see lattigo/utils/buffer/writer.go),
+// it will be wrapped into a [bufio.Writer]. Since this requires allocations, it
+// is preferable to pass a [buffer.Writer] directly:
+//
+//   - When writing multiple times to a [io.Writer], it is preferable to first wrap the
+//     io.Writer in a pre-allocated [bufio.Writer].
+//   - When writing to a pre-allocated var b []byte, it is preferable to pass
+//     buffer.NewBuffer(b) as w (see lattigo/utils/buffer/buffer.go).
+func (evk EvaluationKey) WriteTo(w io.Writer) (n int64, err error) {
+	switch w := w.(type) {
+	case buffer.Writer:
+
+		var inc int64
+		var incInt int
+
+		if inc, err = evk.GadgetCiphertext.WriteTo(w); err != nil {
+			return n + inc, err
+		}
+
+		n += inc
+
+		if evk.IsCompressed() {
+			if incInt, err = w.Write(evk.Seed); err != nil {
+				return n + int64(incInt), err
+			}
+
+			// Sanity check, should not happen unless the size of the seed has been modified in the code
+			if incInt != 32 {
```

Comment:
If this is mandatory, then I would change the seed type to `[32]byte` so that the size is natively enforced.

Reply:
I have a small preference for leaving it as a slice as if the key is not compressed then `seed = nil`. ~This prevents using a nil seed when expanding.~ We should add a check when expanding that `seed != nil`.

Reply:
Then `*[32]byte` will should do it ;)